plugin/src/com/microsoft/alm/plugin/external/commands/EULADialog.java

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root.
+
+package com.microsoft.alm.plugin.external.commands;
+
+import com.intellij.openapi.util.SystemInfo;
+import com.microsoft.alm.helpers.Path;
+import com.microsoft.alm.plugin.external.tools.TfTool;
+import com.microsoft.alm.plugin.external.utils.ProcessHelper;
+import kotlin.text.Charsets;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.JOptionPane;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+import java.awt.Dimension;
+import java.awt.FontMetrics;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import java.util.List;
+import java.util.jar.JarFile;
+
+public class EULADialog {
+    private final static org.slf4j.Logger logger = LoggerFactory.getLogger(EULADialog.class);
+    static volatile boolean wasShown = false;
+    private static final Object myLock = new Object();
+    private static boolean isEULAAccepted = false;
+
+    @Nullable
+    private List<String> getEULAText(){
+        try {
+            String tfLocation = TfTool.getLocation();
+            if (tfLocation == null) {
+                return null;
+            }
+            List<String> result = new ArrayList<String>();
+            String path = Path.combine(Path.combine(Path.getDirectoryName(tfLocation), "lib"), "com.microsoft.tfs.client.common.jar");
+            JarFile jar = new JarFile(path);
```

Comment:
Instead of doing the jar lookup have you considered using an action https://github.com/Microsoft/vso-intellij/tree/master/plugin.idea/src/com/microsoft/alm/plugin/idea/tfvc/actions, rather than looking up the jar?

Reply:
Are you mean call ```tc.cmd eula``` instead of finding a text file in the jar?

Reply:
@leantk Yes, I try to use ```tf eula``` but too complex and unstable.